infrastructure/async/src/main/java/tech/pegasys/teku/infrastructure/async/SafeFuture.java

```diff
@@ -300,6 +327,15 @@ public SafeFuture<T> alwaysRun(final Runnable action) {
         .thenPeek(value -> action.run());
   }
 
+  public void finish(final BiConsumer<T, Throwable> action) {
```

Comment:
I'm not entirely convinced this is a good addition - seems like it would be better to use the separate `finish(success, error)` version.  I can't see anywhere it's actually used either.

Reply:
have to delete this, thanks